### PR TITLE
Add contributions docs

### PR DESCRIPTION
**Pull request by isaacbrodsky**
This PR begins adding some docs about creating bindings, and adds a CONTRIBUTING.md with some other contribution suggestions.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+* Please include tests that show the bug is fixed or feature works as intended.
+* Please add a description of your change to the Unreleased section of the [changelog](./CHANGELOG.md).
+* Please open issues to discuss large features or changes which would break compatibility, before submitting pull requests.
+
+Before we can merge your changes, you must agree to the [Uber Contributor License Agreement](https://cla-assistant.io/uber/h3).
+
+## Other ways to contribute
+
+In addition to making pull requests and Github issues on the @uber/h3 repository, there are other ways you can contribute. Here are some suggestions:
+
```
Comment (dfellis) [inline CONTRIBUTING.md]: Maybe also include improving documentation?

File: dev-docs/creating_bindings.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# Creating bindings for H3
+
+H3 is a C library, in part to make it simpler to create bindings for different programming languages. Each language usually has its own way to bind to C functions, but this document can serve as a starting point and list of tips.
+
+## Function naming
+
+* The `make binding-functions` target produces a file `binding-functions` containing a list of function in the H3 public API, one per line. You can use this as part of your build process to check how much of the H3 public API your bindings expose.
```
Comment (dfellis) [inline dev-docs/creating_bindings.md]: Maybe note somehow that the list doesn't include them memory management functions in H3 that would be useful for other non-memory managed languages, and useful for the binding code in memory managed languages to interface between the two systems.

File: dev-docs/creating_bindings.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# Creating bindings for H3
+
+H3 is a C library, in part to make it simpler to create bindings for different programming languages. Each language usually has its own way to bind to C functions, but this document can serve as a starting point and list of tips.
```
Comment (nrabinowitz) [inline dev-docs/creating_bindings.md]: I might note the bindings that we're intending to release, so that no one wastes time implementing Java, JS, or Go.

Comment (dfellis) [reply dev-docs/creating_bindings.md]: What about Python?

Comment (isaacbrodsky) [reply dev-docs/creating_bindings.md]: Added, even Python. 😛

Comment (isaacbrodsky) [reply dev-docs/creating_bindings.md]: Keep watching #21 :)

**Pull request merged**